puppeteerInterface.js

```diff
@@ -1,5 +1,9 @@
 /** @module puppeteerInterface */
 
+// @ts-nocheck - ruleById property on self Object inside background
```

Comment:
Doesn't this disable it for the entire file, was that intended?

Reply:
Yep, this disables checking for the entire file. It was more trouble than it was worth resolving the `ruleById` issue :(.

Reply:
Cool no worries, I'm mostly concerned in defining the types of the extension integration as that's client code that we'll be interacting with potentially with user generated data etc. :)